src/ansys/motorcad/core/rpc_client_core.py

```diff
@@ -209,6 +224,32 @@ def __init__(
                 + str(self._get_url())
             )
 
+    def _resolve_localhost(self):
+        global SERVER_IP
+
+        ipv6_localhost = "http://[::1]"
+        ipv4_localhost = "http://127.0.0.1"
+
+        if self._check_address_for_response(ipv6_localhost):
+            SERVER_IP = ipv6_localhost
+        elif self._check_address_for_response(ipv4_localhost):
+            SERVER_IP = ipv4_localhost
+        else:
+            SERVER_IP = LOCALHOST_ADDRESS
+
+    def _check_address_for_response(self, address):
```

Comment:
Could you please add a docstring to summarise what this function does, and some comments to explain why we have to set SERVER_IP, how it is used and why we have to restore.